### PR TITLE
HPCC-18800 Improve WebPack out of box experience IV

### PR DESCRIPTION
**Pull request by GordonSmith**
* Simplify ECLWatch development by including a dev webserver (allows 100%
  development from client machine).
* Speed up npm install by including --prefer-offline
* Speed up debug build by skipping compress step
* Option to override compress step:  ECLWATCH_BUILD_STRATEGY=NO_COMPRESS
* Moved some args from package.json to CMakeLists.txt

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq @richardkchapman Please Review

**Comment by AttilaVamos** [comment]
The next Smoketest result won't be correct, based on the ECLWatch won't build in this session. I will deploy a new Smoketest and reschedule this PR soon.

**Comment by GordonSmith** [comment]
@AttilaVamos - Can you rename the "errors" to "lint errors"?

**Comment by AttilaVamos** [comment]
@GordonSmith renamed and it will be active from tomorrow.

File: esp/src/eclwatch/LFDetailsWidget.js [hunk 1]
```diff
@@ -116,6 +116,7 @@ define([
             var context = this;
             var origOnOpen = this.desprayTooltiopDialog.onOpen;
             this.desprayTooltiopDialog.onOpen = function () {
+                var targetRow;
```
Comment (miguelvazq) [inline esp/src/eclwatch/LFDetailsWidget.js]: Is this part of the webpack improvements?

File: esp/src/eclwatch/ESPRequest.js [hunk 2]
```diff
@@ -95,7 +95,7 @@ define([
             return retVal.then(function (response) {
                 params.load(response);
                 return response;
-            }).catch(function(e) {
+            }).catch(function(error) {
```
Comment (miguelvazq) [inline esp/src/eclwatch/ESPRequest.js]: Is this part of the webpack improvements? (I know we moved away from this argument naming convention)

**Comment by GordonSmith** [comment]
Need to add two missing rewrites (only needed for dev mode)

**Comment by miguelvazq** [comment]
The build process requires a `sudo` for TOPOJSON I do not think this should be required.

File: esp/src/eclwatch/ESPRequest.js (hunk 2 above)
Comment (GordonSmith) [reply esp/src/eclwatch/ESPRequest.js]: Its fixing the linting errors

File: esp/src/eclwatch/LFDetailsWidget.js (hunk 1 above)
Comment (GordonSmith) [reply esp/src/eclwatch/LFDetailsWidget.js]: Its fixing the linting errors

**Comment by mckellyln** [comment]
This branch improves package creating time significantly for me from ~11 minutes down to ~3 minutes.

**Comment by GordonSmith** [comment]
@miguelvazq Added missing rewrites to dev server as per your email.
@richardkchapman this PR is ready for pull

**Pull request merged**